jmespath/api.go

```diff
@@ -0,0 +1,24 @@
+package jmespath
+
+// NewLexer creates a new JMESPath lexer.
+func NewLexer() *Lexer {
+	lexer := newLexer()
+	return lexer
+}
+
+// NewParser creates a new JMESPath parser.
+func NewParser() *Parser {
+	parser := newParser()
+	return parser
+}
+
+// Search evaluates a JMESPath expression against input data and returns the result.
+func Search(expression string, data interface{}) (interface{}, error) {
```

Comment:
In addition it would be great if i could provide my own structure for Search to populate instead of returning a new interface that my code would then need to convert and search through.